src/components/core/classes/addClasses.js

```diff
@@ -1,45 +1,38 @@
-export default function addClasses() {
-  const swiper = this;
-  const { classNames, params, rtl, $el, device, support } = swiper;
-  const suffixes = [];
-
-  suffixes.push('initialized');
-  suffixes.push(params.direction);
-
-  if (support.pointerEvents && !support.touch) {
-    suffixes.push('pointer-events');
-  }
-  if (params.freeMode) {
-    suffixes.push('free-mode');
-  }
-  if (params.autoHeight) {
-    suffixes.push('autoheight');
-  }
-  if (rtl) {
-    suffixes.push('rtl');
-  }
-  if (params.slidesPerColumn > 1) {
-    suffixes.push('multirow');
-    if (params.slidesPerColumnFill === 'column') {
-      suffixes.push('multirow-column');
+function pushClasses(settings, containerModifierClass) {
+  const result = [];
+  settings.forEach((item) => {
+    if (typeof item === 'object') {
+      for (const [condition, classNames] of Object.entries(item)) {
```

Comment:
I would avoid using `for of` loops, babel can transpire it to be used as `iterators`. 
```js
Object.entries(item).forEach(([condition, classNames]) => {...}) 
```
will do basically the same